src/main/java/com/feywild/feywild/block/MagicalBrazierBlock.java

```diff
@@ -0,0 +1,141 @@
+package com.feywild.feywild.block;
+
+import com.feywild.feywild.block.entity.MagicalBrazier;
+import com.feywild.feywild.block.render.MagicalBrazierRenderer;
+import io.github.noeppi_noeppi.libx.base.tile.BlockBE;
+import io.github.noeppi_noeppi.libx.mod.ModX;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderers;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.enchantment.EnchantmentHelper;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.RenderShape;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.material.Material;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import javax.annotation.Nonnull;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.function.ToIntFunction;
+
+public class MagicalBrazierBlock extends BlockBE<MagicalBrazier> {
+
+    public static final BooleanProperty BRAZIER_LIT = BooleanProperty.create("brazier_lit");
+    protected static final VoxelShape SHAPE = Block.box(0.0D, 7.0D, 0.0D, 16.0D, 14.0D, 16.0D);
+    private final int FIRE_DAMAGE = 3;
+    private boolean spawnParticles;
```

Comment:
Don't store state in the block class, it is global and breaks stuff. Use a block state property